src/api/src/main/java/engineering/everest/lhotse/api/config/FilterConfig.java

```diff
@@ -0,0 +1,136 @@
+                var organizationId = randomUUID();
+                var registeringUserId = UUID.fromString(accessToken.getSubject());
+                var userEmailAddress = accessToken.getEmail();
+                var organizationName = accessToken.getPreferredUsername();
+                var displayName = otherClaims.getOrDefault(DISPLAY_NAME_KEY, "Guest").toString().trim();
+
+                commandGateway.send(new CreateSelfRegisteredOrganizationCommand(organizationId, registeringUserId,
+                        organizationName, ORGANIZATION_STREET, ORGANIZATION_CITY, ORGANIZATION_STATE,
+                        ORGANIZATION_COUNTRY, ORGANIZATION_POSTAL_CODE, ORGANIZATION_WEBSITE_URL, displayName,
+                        ORGANIZATION_CONTACT_PHONE_NUMBER, userEmailAddress));
+
+                Callable<Boolean> projectionsDone = () -> usersReadService.exists(registeringUserId)
+                        && organizationsReadService.exists(organizationId)
+                        && usersReadService.getById(registeringUserId).getRoles()
+                        .contains(Role.ORG_ADMIN);
+
+                RetryWithExponentialBackoff
+                        .oneMinuteWaiter()
+                        .waitOrThrow(projectionsDone, "user and organization self registration projection update");
+
+                var user = usersReadService.getById(registeringUserId);
+                LOGGER.info("Newly registered user details: " + user);
+
+                // Updating the access token with user claims to avoid token regeneration
+                accessToken.getOtherClaims().putAll(Map.of(ORGANIZATION_ID_KEY, organizationId, DISPLAY_NAME_KEY,
+                        user.getDisplayName(), "roles", user.getRoles()));
+                LOGGER.info("Updated user access token with custom claims.");
+            }
+
+        } catch (Exception e) {
+            LOGGER.error("doFilterInternal error: ", e);
+        } finally {
+            filterChain.doFilter(httpServletRequest, httpServletResponse);
+        }
+    }
+
+    // We are using this method as shouldFilter by doing noneMatch for provided patterns and request paths.
+    // That means, doFilterInternal checks can be applied to only specified includeUrlPatterns.
+    @Override
+    protected boolean shouldNotFilter(HttpServletRequest request) {
+        return Stream.of("/admin/organizations",
+                        "/admin/organizations/**",
+                        "/api/organizations/**",
+                        "/api/organizations/**/users",
+                        "/api/user",
+                        "/api/user/profile-photo",
+                        "/api/user/profile-photo/thumbnail",
+                        "/api/users",
+                        "/api/users/**",
+                        "/api/users/**/forget",
+                        "/api/users/**/roles")
+                .noneMatch(pattern -> new AntPathMatcher().match(pattern, request.getServletPath()));
```

Comment:
These AntPathMatchers are being recreated for every incoming requests which is wasteful.

Try something like:
```suggestion
    private static final Set<String> NOT_INCLUDE_ANT_PATTERNS = Set.of(
            "/admin/organizations",
            "/admin/organizations/**",
            "/api/organizations/**",
            "/api/organizations/**/users",
            "/api/user",
            "/api/user/profile-photo",
            "/api/user/profile-photo/thumbnail",
            "/api/users",
            "/api/users/**",
            "/api/users/**/forget",
            "/api/users/**/roles"
    );

    // We are using this method as shouldFilter by doing noneMatch for provided patterns and request paths.
    // That means, doFilterInternal checks can be applied to only specified includeUrlPatterns.
    @Override
    protected boolean shouldNotFilter(HttpServletRequest request) {
        var pathMatcher = new AntPathMatcher();
        return NOT_INCLUDE_ANT_PATTERNS.stream().noneMatch(pattern -> pathMatcher.match(pattern, request.getServletPath()));
    }
```

Reply:
Done